docs/intro/index.md

```diff
@@ -4,25 +4,92 @@
 
 This is a work in progress.
 
+We'd love to hear your feedback! If you find any errors or have suggestions for
+the book, please [file an
+issue](https://github.com/ahrefs/melange-for-react-devs/issues) or ping @feihong
+on the [#melange channel in the Reason Discord](https://discord.gg/reasonml).
+
 :::
 
-## Motivation
+## What is Melange?
+
+Melange is a collection of tools that bring the robustness of a mature,
+statically-typed multi-paradigm programming language to the JavaScript
+ecosystem. That language is OCaml, a language invented in the 1990s which has
+been battle-tested by industry stalwarts. The heart of Melange is a compiler
+that translates OCaml to human-readable JavaScript and built-in language
+constructs for zero-cost interoperation with JavaScript.
+
+## Why OCaml?
+
+OCaml codebases scale well both in terms of quantity of lines and number of
+contributors. The sound type system helps to prevent ambiguous behavior in your
+program---if it compiles, it runs. OCaml and React (via
+[ReasonReact][reasonreact]) are an effective, FP-friendly combination for
+building frontend UIs using JSX syntax. If the backend is also written in OCaml,
+you can share types between the frontend and backend, ensuring that they stay in
+sync[^1]. In the near future, it will be possible to write [universal React
+components](https://github.com/ml-in-barcelona/server-reason-react) that are
+performantly rendered on the server using OCaml.
+
+All that said, how does OCaml compare against other prominent compile-to-JS
+languages?
+
+**TypeScript** is a gradually-typed superset of JavaScript. It is easier to
+learn and adopt, but is less type-safe than OCaml and all other langues on this
+list. TypeScript doesn't support pattern matching at the language level.
+
+**ReScript** is very similar to OCaml, even at the syntax level. Its
+RescriptReact library is roughly equivalent to our ReasonReact library. The main
+differences are that ReScript's only compilation target is JS, and it lacks the
+metaprogramming features of OCaml, so you lose the ability to, say, [embed CSS
+styles directly into your code](https://styled-ppx.vercel.app/).
+
+**Elm** is similar to OCaml but leans much more strongly towards pure functional
+programming. It cannot use React but comes with its own framework. It has many
+fewer options for JS interop. Even though Elm compiles to JS, it can't be used
+for backend development, even on Node.
 
-This is a project-based, guided introduction to Melange and its ecosystem.
+**F#** is a fullstack language similar to OCaml and is strongly aligned with the
+.NET ecosystem. F# has good support for functional programming and zero-cost
+interop with C#. Feliz is an F# wrapper library for React that doesn't support
+JSX.
+
```

Comment:
I think you can use jsx in Fable/Feliz: https://fable.io/blog/2022/2022-10-12-react-jsx.html , it's just that most people use the elm like syntax.

Reply:
I took a look at JSX in Fable/Feliz, but the usage doesn't seem ergonomic to me (implemented using string templates). Probably just my opinion, but I don't think it qualifies as "JSX".